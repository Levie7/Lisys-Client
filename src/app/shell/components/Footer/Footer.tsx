import * as React from 'react';

import { SettingCompany } from 'src/core/api';

import { Spin } from 'src/shared/components/Spin';
import { getSettings } from 'src/shared/graphql/Setting/schema.gql';
import { queryForm } from 'src/shared/graphql';
import { convertArrayOfObjectsToObject } from 'src/shared/helpers/convertArrayOfObjects';

function FooterPure() {
    let query = queryForm({ query: getSettings, variables: { category: 'company' } });
    if (query.loading) return <Spin />;

    let company = convertArrayOfObjectsToObject(
        query.data?.getSettingsByCategory
    ) as SettingCompany;

    return (
        <div className='bg-primary fg-white p-6 ta-center' id='Footer'>
            Copyright © {company.year} - {company.name}. All Right Reserved.
            <br />
            made with{' '}
            <span role='img' aria-label='Heart'>
                ❤️
            </span>{' '}
            by Lisys
        </div>
    );
}

export const Footer = React.memo(FooterPure);
