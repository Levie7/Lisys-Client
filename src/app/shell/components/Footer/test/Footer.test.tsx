import { ApolloProvider } from '@apollo/react-hooks';
import { act } from '@testing-library/react';
import { mount } from 'enzyme';
import { createMockClient } from 'mock-apollo-client';
import React from 'react';

import { SETTING } from 'src/shared/graphql/Setting/schema.gql';

import { Footer } from '../Footer';

const mockClient = createMockClient();

describe('Footer', () => {
    let wrap: any;
    let queryHandler = jest.fn().mockResolvedValue({
        data: {
            getSettingsByCategory: [
                { type: 'name', value: 'Lisys' },
                { type: 'year', value: '2020' },
            ],
        },
    });
    mockClient.setRequestHandler(SETTING, queryHandler);

    it('should render footer', async () => {
        await act(async () => {
            wrap = mount(
                <ApolloProvider client={mockClient}>
                    <Footer />
                </ApolloProvider>
            );
        });
        expect(wrap.find('Memo(FooterPure)').exists()).toBeTruthy();
    });

    it('should return spin loading', () => {
        expect(wrap.find('Spin').exists()).toBeTruthy();
    });

    describe('when data is loaded', () => {
        beforeEach(async () => {
            await act(async () => {
                wrap = mount(
                    <ApolloProvider client={mockClient}>
                        <Footer />
                    </ApolloProvider>
                );
            });
        });

        it('should return footer with data', () => {
            expect(wrap.find('#Footer').exists()).toBeTruthy();
        });
    });

    it('<Footer/> - should render footer component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });
});
