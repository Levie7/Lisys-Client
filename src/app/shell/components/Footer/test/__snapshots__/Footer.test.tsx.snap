// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer <Footer/> - should render footer component correctly 1`] = `ReactWrapper {}`;
